NightscoutUploadKit/TimeFormat.swift

```diff
@@ -9,13 +9,18 @@
 import Foundation
 
 class TimeFormat: NSObject {
-    private static var formatterISO8601 = DateFormatter.ISO8601DateFormatter()
+    private static var formatterISO8601 = ISO8601DateFormatter.defaultFormatter()
     
     static func timestampStrFromDate(_ date: Date) -> String {
         return formatterISO8601.string(from: date)
     }
 
     static func dateFromTimestamp(_ string: String) -> Date? {
-        return formatterISO8601.date(from: string)
+        if let result = formatterISO8601.date(from: string) {
+            return result
+        }
+            
+        //Nightscout is returning this format in some cases...  needs more research
+        return ISO8601DateFormatter.fractionalSecondsFormatter().date(from: string)
```

Comment:
Constructing formatters can be very expensive, involving locales, calendars, etc. Recommend to keep the construction of the formatter a singleton of sorts.

Reply:
Good idea and normally I do use static but didn't think of it this time. I updated it to follow the paradigm in this class for the existing formatter here and made a private reference in this class.